From 0f0b4b7620557410e4969cec6119a7771b8acf35 Mon Sep 17 00:00:00 2001
From: Omar Polo <op@omarpolo.com>
Date: Mon, 11 Jul 2022 11:52:43 +0200
Subject: [PATCH] fix gzdoom.pk3 not found error

Move the initialization before BaseFileSearch is called, otherwise
GameConfig is used not initialized and it doesn't find the gzdoom.pk3
file.

GameConfig used uninitalized was spotted by @LoneFox78.
---
 src/d_main.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/d_main.cpp b/src/d_main.cpp
index 46dbe0f33..df1051af5 100644
--- a/src/d_main.cpp
+++ b/src/d_main.cpp
@@ -3534,6 +3534,8 @@ static int D_DoomMain_Internal (void)
 	
 	std::set_new_handler(NewFailure);
 	const char *batchout = Args->CheckValue("-errorlog");
+
+	D_DoomInit();
 	
 	// [RH] Make sure zdoom.pk3 is always loaded,
 	// as it contains magic stuff we need.
@@ -3568,8 +3570,6 @@ static int D_DoomMain_Internal (void)
 
 	Printf("%s version %s\n", GAMENAME, GetVersionString());
 
-	D_DoomInit();
-
 	extern void D_ConfirmSendStats();
 	D_ConfirmSendStats();
 
-- 
2.37.2

